pkg/svcat/service-catalog/sdk.go

```diff
@@ -17,12 +17,66 @@ limitations under the License.
+	BindingParentHierarchy(*apiv1beta1.ServiceBinding) (*apiv1beta1.ServiceInstance, *apiv1beta1.ClusterServiceClass, *apiv1beta1.ClusterServicePlan, *apiv1beta1.ClusterServiceBroker, error)
+	DeleteBinding(string, string) error
+	DeleteBindings([]types.NamespacedName) ([]types.NamespacedName, error)
+	IsBindingFailed(*apiv1beta1.ServiceBinding) bool
+	IsBindingReady(*apiv1beta1.ServiceBinding) bool
+	RetrieveBinding(string, string) (*apiv1beta1.ServiceBinding, error)
+	RetrieveBindings(string) (*apiv1beta1.ServiceBindingList, error)
+	RetrieveBindingsByInstance(*apiv1beta1.ServiceInstance) ([]apiv1beta1.ServiceBinding, error)
+	Unbind(string, string) ([]types.NamespacedName, error)
+	WaitForBinding(string, string, time.Duration, *time.Duration) (*apiv1beta1.ServiceBinding, error)
+
+	RetrieveBrokers() ([]apiv1beta1.ClusterServiceBroker, error)
+	RetrieveBroker(string) (*apiv1beta1.ClusterServiceBroker, error)
+	RetrieveBrokerByClass(*apiv1beta1.ClusterServiceClass) (*apiv1beta1.ClusterServiceBroker, error)
+	Register(string, string) (*apiv1beta1.ClusterServiceBroker, error)
+	Sync(string, int) error
+
+	RetrieveClasses() ([]apiv1beta1.ClusterServiceClass, error)
+	RetrieveClassByName(string) (*apiv1beta1.ClusterServiceClass, error)
+	RetrieveClassByID(string) (*apiv1beta1.ClusterServiceClass, error)
+	RetrieveClassByPlan(*apiv1beta1.ClusterServicePlan) (*apiv1beta1.ClusterServiceClass, error)
+
+	Deprovision(string, string) error
+	InstanceParentHierarchy(*apiv1beta1.ServiceInstance) (*apiv1beta1.ClusterServiceClass, *apiv1beta1.ClusterServicePlan, *apiv1beta1.ClusterServiceBroker, error)
+	InstanceToServiceClassAndPlan(*apiv1beta1.ServiceInstance) (*apiv1beta1.ClusterServiceClass, *apiv1beta1.ClusterServicePlan, error)
+	IsInstanceFailed(*apiv1beta1.ServiceInstance) bool
+	IsInstanceReady(*apiv1beta1.ServiceInstance) bool
+	Provision(string, string, string, string, string, interface{}, map[string]string) (*apiv1beta1.ServiceInstance, error)
+	RetrieveInstance(string, string) (*apiv1beta1.ServiceInstance, error)
+	RetrieveInstanceByBinding(*apiv1beta1.ServiceBinding) (*apiv1beta1.ServiceInstance, error)
+	RetrieveInstances(string, string, string) (*apiv1beta1.ServiceInstanceList, error)
+	RetrieveInstancesByPlan(*apiv1beta1.ClusterServicePlan) ([]apiv1beta1.ServiceInstance, error)
+	TouchInstance(string, string, int) error
+	WaitForInstance(string, string, time.Duration, *time.Duration) (*apiv1beta1.ServiceInstance, error)
+
+	RetrievePlans(*FilterOptions) ([]apiv1beta1.ClusterServicePlan, error)
+	RetrievePlanByName(string) (*apiv1beta1.ClusterServicePlan, error)
+	RetrievePlanByID(string) (*apiv1beta1.ClusterServicePlan, error)
+	RetrievePlansByClass(*apiv1beta1.ClusterServiceClass) ([]apiv1beta1.ClusterServicePlan, error)
+	RetrievePlanByClassAndPlanNames(string, string) (*apiv1beta1.ClusterServicePlan, error)
+
+	RetrieveSecretByBinding(*apiv1beta1.ServiceBinding) (*apicorev1.Secret, error)
+
+	ServerVersion() (*version.Info, error)
+}
```

Comment:
Not being sure how this interface is going to be used, I wonder if it best split up into multiple sub interfaces, approximately how it is split by newlines.

Reply:
And then what? Have the Svcat app type inline multiple interfaces? Seems like it would be more of a mess to fake, esp. considering you would have to update them individually with Counterfeiter. 

Having one bighuge interface does seem kind of messy, but I don't think splitting it up is actually a better alternative.

Reply:
Exactly so. If this isn't intended to be used as a nicer version of a client in any way, then we can leave it, otherwise I think it would be nice to have a binder and a provisioner, or a broker client and a plan client, etc.
I'm fine with it as is. Like I said, I haven't thought about how the interface is to be used.

Reply:
I am fine with a mega interface for now. If there's a need to break it down further, let's tackle that then.